kext_1.10.go

```diff
@@ -0,0 +1,118 @@
+// +build darwin,!ios
+// +build go1.10
+
+package kext
+
+/*
+#cgo LDFLAGS: -framework CoreFoundation -framework IOKit
+
+#include <IOKit/kext/KextManager.h>
+*/
+import "C"
+import "fmt"
+
+type Info struct {
+	Version string
+	Started bool
+}
+
+func LoadInfo(kextID string) (*Info, error) {
+	info, err := LoadInfoRaw(kextID)
+	if err != nil {
+		return nil, err
+	}
+	if info == nil {
+		return nil, nil
+	}
+
+	return &Info{
+		Version: info["CFBundleVersion"].(string),
+		Started: info["OSBundleStarted"].(bool),
+	}, nil
+}
+
+func LoadInfoRaw(kextID string) (map[interface{}]interface{}, error) {
+	cfKextID, err := StringToCFString(kextID)
+	if cfKextID != 0 {
+		defer Release(C.CFTypeRef(cfKextID))
+	}
+	if err != nil {
+		return nil, err
+	}
+	cfKextIDs := ArrayToCFArray([]C.CFTypeRef{C.CFTypeRef(cfKextID)})
+	if cfKextIDs != 0 {
+		defer Release(C.CFTypeRef(cfKextIDs))
+	}
+
+	cfDict := C.KextManagerCopyLoadedKextInfo(C.CFArrayRef(cfKextIDs), 0)
```

Comment:
This cast shouldn't be necessary, since `ArrayToCFArray` returns a `C.CFArrayRef`, but it is. (There are similar cases below.) I'll file a bug upstream.

Reply:
Bug filed: https://github.com/golang/go/issues/24161